pkg/apis/postgres-operator.crunchydata.com/v1beta1/shared_types.go

```diff
@@ -311,14 +311,18 @@ func (meta *Metadata) GetAnnotationsOrNil() map[string]string {
 // Only one applier should be managing each volume definition.
 // https://docs.k8s.io/reference/using-api/server-side-apply#merge-strategy
 // +structType=atomic
+//
+// +kubebuilder:validation:XValidation:rule=`has(self.claimName) != has(self.image)`,message=`you must set only one of image or claimName`
+// +kubebuilder:validation:XValidation:rule=`!has(self.image) || !has(self.readOnly) || self.readOnly`,message=`readOnly cannot be set false when using an ImageVolumeSource`
+// +kubebuilder:validation:XValidation:rule=`!has(self.image) || (self.?image.reference.hasValue() && self.image.reference.size() > 0)`,message=`if using an ImageVolumeSource, you must set a reference`
```

Comment:
😩 So the upstream type lacks `+required` and instead puts `Required:` in the description, eh?

Reply:
And the comment describes that it's not required because it might be overwritten (by the spec it's in?), which doesn't quite explain things to me. I mean, I guess it behaves somewhat like a RELATED_ image field in OLM.